wandb/sdk/artifacts/storage_handlers/s3_handler.py

```diff
@@ -285,10 +285,11 @@ def _etag_from_obj(
         ]
     ) -> ETag:
         etag: ETag
-        if hasattr(obj, "e_tag"):
+        boto = util.get_module("boto3")
+        try:
             etag = obj.e_tag[1:-1]  # escape leading and trailing quote
-        else:
-            etag = obj.get()["ETag"][1:-1]  # escape leading and trailing quote
+        except boto.exceptions.ResourceLoadException:
```

Comment:
lets not use exceptions for regular control flow: https://docs.python.org/2/faq/design.html#how-fast-are-exceptions the cost of handling an exception is very expensive and we expect this code path to be execute frequently when downloading a large reference versioned artifact